src/NuGet.Indexing/DownloadsByVersion.cs

```diff
@@ -48,8 +41,13 @@ public int Total
 
             set
             {
+                if (_downloadsByVersion.ContainsKey(version))
+                {
+                    _total = _total - _downloadsByVersion[version];
```

Comment:
It is only out of sync while the data is being reloaded. And looking further up the call tree into the usage of this object during reload, there is no assumption that the processed data is everything. In fact, the only assumption is that the incoming json file that we are "reloading" contains some download data and will only overwrite previous loads where they overlap. Currently, in practice, I don't THINK we have any cases where the new downloads isn't a superset of the previous one, but without confirmation, I'm hesitant to change it.